src/core/Akka.Persistence/Journal/AsyncWriteJournal.cs

```diff
@@ -330,33 +345,11 @@ private void HandleWriteMessages(WriteMessages message)
             var self = Self;
             _resequencerCounter += message.Messages.Aggregate(1, (acc, m) => acc + m.Size);
             var atomicWriteCount = message.Messages.OfType<AtomicWrite>().Count();
-            AtomicWrite[] prepared;
-            Task<IImmutableList<Exception>> writeResult;
-            Exception writeMessagesAsyncException = null;
-            try
-            {
-                prepared = PreparePersistentBatch(message.Messages).ToArray();
-                // try in case AsyncWriteMessages throws
-                try
-                {
-                    writeResult = _breaker.WithCircuitBreaker(() => WriteMessagesAsync(prepared));
-                }
-                catch (Exception e)
-                {
-                    writeResult = Task.FromResult((IImmutableList<Exception>) null);
-                    writeMessagesAsyncException = e;
-                }
-            }
-            catch (Exception e)
-            {
-                // exception from PreparePersistentBatch => rejected
-                writeResult = Task.FromResult((IImmutableList<Exception>)Enumerable.Repeat(e, atomicWriteCount).ToImmutableList());
-            }
 
             void Resequence(Func<IPersistentRepresentation, Exception, object> mapper, IImmutableList<Exception> results)
```

Comment:
IDK if it's worth doing, but should we ValueDelegate Here? i.e.

```
//Declared elsewhere
public interface IResequenceMapper:
{
   public object Invoke(IPersistentRepresentation rep, Exception ex);
}

void Resequence<TMapper>(TMapper, IImmutableList<Exception> results)
where TMapper: struct, IResequenceMapper
```

This will get rid of the cost of indirection from calling `Func`s, with the tradeoff of slightly larger code (Due to each mapper getting its own version of `Resequence`.

Reply:
Absolutely - I had made a mental note about "how do I get rid of these two delegate allocations?" inside this method and a value delegate is probably the simplest route.